frontend/src/components/nodes/logic/CoalesceNode.tsx

```diff
@@ -0,0 +1,385 @@
+import React, { useState, useRef, useEffect, useMemo } from 'react';
+import { Handle, Position, useConnection } from '@xyflow/react';
+import BaseNode from '../BaseNode';
+import {
+  Input,
+  Card,
+  Divider,
+  Button,
+  Select,
+  SelectItem
+} from '@nextui-org/react';
+import { useDispatch, useSelector } from 'react-redux';
+import { updateNodeData } from '../../../store/flowSlice';
+import styles from '../DynamicNode.module.css';
+import { Icon } from '@iconify/react';
+import { RootState } from '../../../store/store';
+import NodeOutputDisplay from '../NodeOutputDisplay';
+import isEqual from 'lodash/isEqual';
+import { FlowWorkflowNode } from '../../../store/flowSlice';
+
+interface CoalesceNodeData {
+  color?: string;
+  config: {
+    preferences: string[];
+    input_schema?: Record<string, string>;
+    output_schema?: Record<string, string>;
+    title?: string;
+  };
+  run?: Record<string, any>;
+  taskStatus?: string;
+}
+
+interface CoalesceNodeProps {
+  id: string;
+  data: CoalesceNodeData;
```

Comment:
This can be `FlowWorkflowNode['data']`
We should use already declared types as much as possible.
No need to declare CoalesceNodeData type separately.